pkg/testutils/mockprovider/mock_kube_provider.go

```diff
@@ -0,0 +1,40 @@
+package mockprovider
```

Comment:
Can we use `go:generate` for this?

Reply:
This provides an implementation that returns a statically created client, not just stubbed fields. If it should go elsewhere (or be named something other than 'mock' to distinguish between them) I'm all ears.